### ISSUE TITLE
Simple AssignmentInCondition?

### ISSUE THREAD

**Issue by scottwater** [testdouble/standard#65]
Not a deal breaker, but since it is pre-release....

```ruby
if post = Post.find_by(id: id)

end
```
Generates a warning about AssignmentInCondition. The fix is to just wrap it in parentheses, but i just hate the look of 

```ruby
if (post = Post.find_by(id: id))

end
```

For now, I will likely just add an extra line:
```ruby
post = Post.find_by(id: id)
if post

end
```

I would love to see an option to allow simple assignments to pass.

Thanks,
Scott

**Comment by garybernhardt** [comment]
Earlier this year, I decided to permanently disallow `if x = y` in all of my projects. The reason was that we had an actual production issue where someone wrote `if x = y`, but meant `if x == y`. Neither the tests nor the author nor the reviewer caught it. I used Ruby for a decade before seeing such a defect make it to production, but it finally did. But even if it's infrequent, it's not worth that risk just to avoid splitting one line into two and naming a variable.

I think that's the argument that I made to @searls when convincing him to add this rule to standard.
